src/DataLayouts/cuda.jl

```diff
@@ -76,3 +76,58 @@ function Base.copyto!(
     CUDA.@cuda threads = (Nij, Nij) blocks = (Nh,) knl_copyto!(dest, bc)
     return dest
 end
+
+function knl_mapreduce!(dest, fn, op, src)
```

Comment:
Is the intent to write a generic `mapreduce` function or do we want to focus on a restricted set of `op`s used with the operators?